[Hotfix 4.0.1] | Suppress TLS security warning with Encrypt=false by new AppContext switch (#1467)
dotnet/SqlClient: BUILDGUIDE.md

`Switch.Microsoft.Data.SqlClient.EnableSecureProtocolsByOS`

## Suppressing TLS security warning

> When connecting to a server, if a protocol lower than TLS 1.2 is negotiated, a security warning is output to the console. This warning can be suppressed on SQL connections with `Encrypt = false` by enabling the following AppContext switch on application startup:

Reviewer: ```suggestion
When connecting to a Sql server, if a protocol lower than TLS 1.2 is negotiated, a security warning is printed to the console. This warning can be suppressed on SQL connections while `Encrypt = false` by enabling the following AppContext switch on application startup:
```

Author: This is a backport, and we usually keep them identical with the source on the main branch. You can prepare a PR on the main branch If you think we need improvement here and backport it after getting merged.
